task_binarysearch/js/main.js

```diff
@@ -0,0 +1,70 @@
+"use strict";
+
+var array = [-5, -2, 2, 3, 5, 7, 8, 10, 12, 15, 30, 40, 90, 120];
+var value = 90;
+var timers = {};
+var BINARY_SEARCH = "binarySearch";
+var BINARY_SEARCH_RECURS = "binarySearchRecursion";
+binarySearch = timingDecorator(binarySearch, BINARY_SEARCH);
+binarySearchRecursion = timingDecorator(binarySearchRecursion, BINARY_SEARCH_RECURS);
+logBinarySearch(array, value, binarySearch(array, value), BINARY_SEARCH,
+        timers[BINARY_SEARCH]);
+logBinarySearch(array, value, binarySearchRecursion(array, value), BINARY_SEARCH_RECURS,
+        timers[BINARY_SEARCH_RECURS]);
+
+function binarySearch(array, value) {
+  var index = -1;
+  var low = 0, high = array.length, middle;
+  while (low < high) {
+    middle = low + (high - low >> 1);
+    if (value === array[middle]) {
+      index = middle;
+      break;
+    } else {
+      if (value < array[middle]) {
+        high = middle;
+      } else {
+        low = middle + 1;
+      }
+    }
+  }
+  return index;
+}
+
+function binarySearchRecursion(array, value) {
+  var index = -1;
+  var low = arguments[2] || 0;
+  var high = arguments[3] || array.length;
```

Comment:
it can be a mistake when `arguments[3]` is passed as 0.
0 || 5 = 5
but function works, I suppose because of binary operator
please fix this moment, thanks